src/picosvg/svg_types.py

```diff
@@ -519,11 +536,18 @@ def as_path(self) -> SVGPath:
         return path
 
 
+def _bool_op_clip_paths(shapes: Iterable[SVGShape]) -> Generator[SVGPath, None, None]:
+    for shape in shapes:
+        path = shape.as_cmd_seq()
+        if shape.clip_rule == "evenodd":
+            yield path.remove_overlaps(fill_rule="evenodd", inplace=True)
+        else:
+            yield path
```

Comment:
You assume the return type of as_cmd_seq in an unsafe manner; if we were typed this wouldn't compile.

```
for shape in shapes:
  path = shape.as_path()
  if shape.clip_rule == "evenodd":
    path = path.remove_overlaps
  yield path.as_cmd_seq
```

Return type should be SVGCommandSeq not SVGPath.

Reply:
remove_overlaps() method itself already calls as_cmd_seq() on self before passing control to svg_pathos, calling again as_cmd_seq() on the result of remove_overlaps seemed unnecessary to me..